specs/IsNonClientRegionSupportEnabled.md

```diff
@@ -0,0 +1,176 @@
+
+## Win32 C++
+```cpp 
+ScenarioNonClientRegionSupport::ScenarioNonClientRegionSupport(AppWindow* appWindow)
+    : m_appWindow(appWindow), m_webView(appWindow->GetWebView())
+{
+    
+    CHECK_FAILURE(m_webView->add_NavigationStarting(
+        Callback<ICoreWebView2NavigationStartingEventHandler>(
+            [this](
+                ICoreWebView2* sender,
+                ICoreWebView2NavigationStartingEventArgs* args) -> HRESULT
+            {
+                static const PCWSTR allowedHostName = L"www.microsoft.com";
+                wil::unique_cotaskmem_string uri;
+                CHECK_FAILURE(args->get_Uri(&uri));
+                wil::unique_bstr domain = GetDomainOfUri(uri.get());
+                
+                wil::com_ptr<ICoreWebView2Settings> m_settings;
+                CHECK_FAILURE(m_webView->get_Settings(&m_settings));
+                wil::com_ptr<ICoreWebView2Settings12> coreWebView2Settings12;
+                coreWebView2Settings12 = m_settings.try_query<ICoreWebView2Settings12();
+                CHECK_FEATURE_RETURN(coreWebView2Settings12);
+                
+                BOOL enabled;
+                CHECK_FAILURE(coreWebView2Settings12->get_IsNonClientRegionSupportEnabled(&enabled));
+
+                if (_wcsicmp(domain.get(), allowedHostName) == 0 && !enabled)
+                {
+                    CHECK_FAILURE(
+                        coreWebView2Settings12->put_IsNonClientRegionSupportEnabled(TRUE));
+                }
+                else if (_wcsicmp(domain.get(), allowedHostName) != 0 && enabled)
+                {
+                    CHECK_FAILURE(
+                        coreWebView2Settings12->put_IsNonClientRegionSupportEnabled(FALSE));
+                }
+                return S_OK;
+            })
+            .Get(),
+        &m_navigationStartingToken));
+}
+```
+## .NET C#
+```c#
+// WebView2 control is defined in the xaml
+// <wv2:WebView2 x:Name="webView" Source="https://www.microsoft.com/"/>
+public MainWindow() 
+{
+    InitializeComponent();
+    webView.NavigationStarting += SetNonClientRegionSupport;
+}
+
+private void SetNonClientRegionSupport(CoreWebView2 sender, CoreWebView2NavigationStartingEventArgs args)
+{
+    var coreWebView2Settings = webView.CoreWebView2.Settings;
+    var urlCompareExample = "www.microsoft.com";
+    var uri = new Uri(args.Uri);
+
+    if (String.Equals(uri.Host, urlCompareExample, StringComparison.OrdinalIgnoreCase) &&
+        !coreWebView2Settings.IsNonClientRegionSupportEnabled)
+    {
+        coreWebView2Settings.IsNonClientRegionSupportEnabled = true;
+    }
+    else if (!String.Equals(uri.Host, urlCompareExample, StringComparison.OrdinalIgnoreCase) && 
+        coreWebView2Settings.IsNonClientRegionSupportEnabled)
+    {
+        coreWebView2Settings.IsNonClientRegionSupportEnabled = false;
+    }
+}
+```
+
+## Declaring Non-client App Regions
+Non-client regions are HTML elements that are marked with the css style `app-region`.
```

Comment:
Statement isn't actually true. The presence of `app-region` doesn't make you non-client. Rather, then value of the `app-region` attribute controls whether you are non-client. Suggest something like

> The `app-region` CSS property is used to indicate which HTML elements are treated as non-client regions.

Reply:
Please fix.

Reply:
fixed